src/docs/ad/ad-service-discovery.rst

```diff
@@ -0,0 +1,84 @@
+Service Discovery
+#################
+
+Active Directory Terms
+======================
+
+`Active Directory <https://en.wikipedia.org/wiki/Active_Directory>`_ is a directory service providing Windows and Linux clients with information about centrally stored user and group information. Below are some Active Directory terms that are important to understand for the purposes of this document.
+
+Domain Controller
+    The Active Directory Domain Controller (DC for short) is an important source of user information and authentication in many computing environments. The DC stores user and group information in a LDAP directory server (database) and provides this information as a `service <https://docs.microsoft.com/en-us/windows-server/identity/ad-ds/get-started/virtual-dc/active-directory-domain-services-overview>`_. This allows administrators to configure clients to retrieve user information from the AD DC. Similarly, the DC contains Kerberos data used to authenticate users password and access to other network services.
+
+Domain
+    A domain is defined as a logical group of network objects (computers, users, devices) that share the same Active Directory database.
+
+Forest
+    A forest is a collection of trees that share a common global catalog, directory schema, logical structure, and directory configuration. A forest contains one or multiple domains.
+
+Site
+    A logical organization, or container for objects within a domain. `Sites <https://docs.microsoft.com/en-us/windows-server/identity/ad-ds/plan/understanding-active-directory-site-topology>`_ are often used to group geographically-close systems together. Clients can prefer to communicate only to servers within a site.
+
+Global Catalog
+    A service advertising information about Active Directory domains in a forest, it can be searched by clients similarly to normal LDAP user/group information. The `Global Catalog <https://docs.microsoft.com/en-us/windows/win32/ad/global-catalog>`_ is used by SSSD clients to read AD Site and Forest information, and trusted domains.
```

Comment:
The netlogon attribute that is used to get site and forest names is present on every DC and we look it up in DC not GC. Some domain controllers are global catalogs, not all domain controllers are global catalogs. Global catalog is a little bit more difficult thing, we use it to lookup user and group information (not the site and forest). Perhaps it is not needed to mention it here and complicate stuff.

Reply:
It does a lot more then advertising information about domains which is not needed for the sake of this article, so I'd vote to remove it. In short, global catalog contains information (selected attributes) of every object in the forest (from all domains).

Reply:
Okay makes sense, I removed 'global catalog' from the terminology list.